app/src/main/java/ch/epfl/sdp/appart/database/firestoreservicehelpers/FirestoreAdHelper.java

```diff
@@ -340,6 +340,24 @@ public FirestoreAdHelper() {
         return result;
     }
 
+    public CompletableFuture<Boolean> deleteAd(String adId, String cardId) {
+        CompletableFuture<Boolean> finalResult = new CompletableFuture<>();
+
+        CompletableFuture<Boolean> result1 = new CompletableFuture<>();
+
+        db.collection(adsPath).document(adId).delete().addOnCompleteListener(t -> {
+            if (t.isSuccessful())
+                result1.complete(true);
+            else
+                result1.complete(false);
```

Comment:
It would be better to keep the same structure we used on other cases: use a Void future that completes normally if the task is successful, and completes exceptionally with a `DatabaseServiceException` containing the task failure message. Same goes for `finalResult`.

Reply:
Makes sense, I'll change that !

Reply:
Now looking back to what was done in the other, the only methods that seem to return a Void future are the private methods. In DatabaseService though, all the public methods return rather a CompletableFuture<Boolean> (including deleteImage(), which I think is the most similar to the method here). Then shouldn't deleteAd also be a CompletableFuture<Boolean> ?